### DISCUSSION TITLE
Filter 错误处理之后的返回值，如何全局统一拦截

### DISCUSSION THREAD

**Discussion by willjayyyy** [midwayjs/midway#3964] [Q&A]
### Describe the problem（描述问题）

有这么个场景，需要对 Filter 中 catch 到的错误，会返回特定的 http 状态码及返回值，目前 middleware 和@Match 均无法捕获@Catch 的返回值，有什么方法可以全局统一拦截吗

### Midway Versions（Midway 版本）

✓ @midwayjs/faas-typings(not installed)
✓ @midwayjs/fc-starter(not installed)
✓ @midwayjs/serverless-http-parser(not installed)
✓ @midwayjs/async-hooks-context-manager(not installed)
✓ @midwayjs/axios(3.16.5)
✓ @midwayjs/bootstrap(3.16.2)
✓ @midwayjs/bull(3.16.5)
✓ @midwayjs/bull-board(not installed)
✓ @midwayjs/cache-manager(not installed)
✓ @midwayjs/captcha(not installed)
✓ @midwayjs/casbin(not installed)
✓ @midwayjs/casbin-redis-adapter(not installed)
✓ @midwayjs/casbin-typeorm-adapter(not installed)
✓ @midwayjs/code-dye(not installed)
✓ @midwayjs/consul(not installed)
✓ @midwayjs/core(3.16.2)
✓ @midwayjs/cos(not installed)
✓ @midwayjs/cron(not installed)
✓ @midwayjs/cross-domain(not installed)
✓ @midwayjs/decorator(not installed)
✓ @midwayjs/etcd(not installed)
✓ @midwayjs/express-session(not installed)
✓ @midwayjs/faas(not installed)
✓ @midwayjs/grpc(not installed)
✓ @midwayjs/http-proxy(not installed)
✓ @midwayjs/i18n(not installed)
✓ @midwayjs/info(3.16.5)
✓ @midwayjs/jwt(not installed)
✓ @midwayjs/kafka(not installed)
✓ @midwayjs/leoric(not installed)
✓ @midwayjs/mikro(not installed)
✓ @midwayjs/mock(3.16.5)
✓ @midwayjs/mongoose(not installed)
✓ @midwayjs/mqtt(not installed)
✓ @midwayjs/oss(not installed)
✓ @midwayjs/otel(not installed)
✓ @midwayjs/passport(not installed)
✓ @midwayjs/process-agent(not installed)
✓ @midwayjs/prometheus(not installed)
✓ @midwayjs/prometheus-socket-io(not installed)
✓ @midwayjs/rabbitmq(not installed)
✓ @midwayjs/redis(3.16.5)
✓ @midwayjs/security(not installed)
✓ @midwayjs/sequelize(not installed)
✓ @midwayjs/session(3.16.5)
✓ @midwayjs/socketio(not installed)
✓ @midwayjs/static-file(not installed)
✓ @midwayjs/swagger(not installed)
✓ @midwayjs/tablestore(not installed)
✓ @midwayjs/tags(not installed)
✓ @midwayjs/tenant(not installed)
✓ @midwayjs/typegoose(3.16.5)
✓ @midwayjs/typeorm(not installed)
✓ @midwayjs/upload(3.16.5)
✓ @midwayjs/validate(3.16.5)
✓ @midwayjs/version(not installed)
✓ @midwayjs/view(not installed)
✓ @midwayjs/view-ejs(not installed)
✓ @midwayjs/view-nunjucks(not installed)
✓ @midwayjs/web(not installed)
✓ @midwayjs/express(not installed)
✓ @midwayjs/koa(3.16.5)
✓ @midwayjs/ws(not installed)

**Comment by czy88840616** [comment]
Filter 不就是做这个事情的么。。。

**Reply by willjayyyy** [reply]
不同的 filter 有不同的返回结果，我希望把正常 router 和 filter 的返回值（也就是接口的返回值）进行统一处理，如格式转换等，但是目前来看 filter 是最外层了，无法在 filter 外再拦截一层

**Reply by czy88840616** [reply]
是的，filter已经是业务兜底了，再包一层又是套娃了。
